### PR TITLE
Introduce a group for collections

### PR DESCRIPTION
**Pull request by josevalim**
Currently the docs have a large group called "Data & Behaviour". The goal of this PR is to extract "Collections" (lists, sets, etc) and "Enumerables" (such as ranges) into their own group. Feedback is welcome!

### CODE REVIEW

File: lib/elixir/docs.exs [hunk 1]
```diff
@@ -4,36 +4,39 @@
...
-      Map,
-      MapSet,
       NaiveDateTime,
-      Range,
       Record,
       Regex,
-      Stream,
       String,
       Time,
       Tuple,
```
Comment (michalmuskala) [inline lib/elixir/docs.exs]: Should `Tuple` also live in collections?

**Comment by michalmuskala** [comment]
I think this is a move in the right direction, I'm not particularly convinced by the name "Collections & Enumerables" - I think "Collections" or "Collection Types" would be enough. You could easily view range as an optimised collection of sequential integers.

File: lib/elixir/docs.exs (hunk 1 above)
Comment (josevalim) [reply lib/elixir/docs.exs]: I would say Tuple is a composite type but not a collection.

**Comment by josevalim** [comment]
> You could easily view range as an optimised collection of sequential integers.

I like Collections if we can agree with the above.

**Comment by ianrumford** [comment]
The above discussion just highlights my point :-)  There is no "solution" and all groupings are essentially arbitrary.

A while ago I opened [#6571](https://github.com/elixir-lang/elixir/pull/6571) regarding the removal of the distinction between "functions" from "macros" in Kernel.SpecialForms.  Jose made the point that newcomers found that confusing.  I accepted that.

But adding these groups are even more confusing for newcomers in my view  and I can't reconcile keeping it simple for newcomers v adding these groups which really only make sense to people who already know their way around the the language.

Which goes to the heart of who is the target audience for the reference documentation? If its newcomers (and I think that's best) then the simpler the better.

**Comment by josevalim** [comment]
> But adding these groups are even more confusing for newcomers in my view

As I said in #7184, we need to validate this. Otherwise, we will only be guessing which one is better.

**Comment by Awenimpagaritse** [comment]
Cool

File: lib/elixir/docs.exs (hunk 1 above)
Comment (Awenimpagaritse) [inline lib/elixir/docs.exs]: Cool

Comment (Awenimpagaritse) [reply lib/elixir/docs.exs]: Cool

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

File: lib/elixir/docs.exs (hunk 1 above)
Comment (fertapric) [reply lib/elixir/docs.exs]: The first paragraph of the `Tuple` documentation says: [`Tuples are ordered collections of elements`](https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/tuple.ex#L5). Should we update the documentation expressing is a composite type?

The [Erlang documentation](http://erlang.org/doc/reference_manual/data_types.html) says:

> A tuple is a compound data type with a fixed number of terms:
>
> `{Term1,...,TermN}`
>
> Each term `Term` in the tuple is called an *element*. The number of elements is said to be the size of the tuple.

**Comment by josevalim** [comment]
Good call, we should update it.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Pull request merged**